torch/csrc/jit/tracer.h

```diff
@@ -69,6 +71,24 @@ inline bool isTracing(const variable_list& vars) {
   return false;
 }
 
+inline bool isTracing(const at::TensorList& vars) {
+  for (Variable var : vars) {
+    if (!var.defined() || !var.tracing_state()) continue;
+    if (std::any_of(var.tracing_state()->begin(), var.tracing_state()->end(), detail::isElemActive))
+      return true;
+  }
+  return false;
+}
+
+inline bool isTracing(const at::ArrayRef<Variable>& vars) {
+  for (const Variable& var : vars) {
+    if (!var.defined() || !var.tracing_state()) continue;
+    if (std::any_of(var.tracing_state()->begin(), var.tracing_state()->end(), detail::isElemActive))
+      return true;
+  }
+  return false;
+}
```

Comment:
I see two options to deduplicate them:
1. template `isTracing` on argument type
2. put the logic in an overload that gets begin and end iterators and call this from all overloads you implemented here

Reply:
I don't think templating quite works; iterators is a good idea though.

Reply:
I'd be surprised if it didn't. It should be ok

Reply:
You can't factor the body of the loop, it uses a short-circuiting `return`.

Reply:
It turns out iterators don't work either.

Templating doesn't work because in all cases except TensorList, you want to take a reference to avoid refcount bump; but for TensorList, we need to implicit construct a Variable to get the types to line up.